Firestore/core/src/util/async_queue.cc

```diff
@@ -86,6 +86,7 @@ void AsyncQueue::ExecuteBlocking(const Operation& operation) {
               "ExecuteBlocking may not be called "
               "before the previous operation finishes executing");
 
+  auto self = shared_from_this();
```

Comment:
Is there a reason we haven't done this before? `is_operation_in_progress_ = false;` may crash when `api::Firestore` (and, consequently, the worker queue) is destroyed on the queue (in `FirestoreClient::TerminateAsync`).

Reply:
This seems like an oversight.

I think the idea at the time was that FirestoreClient would keep the AsyncQueue alive as long as it needed to, but this doesn't account for the case where the Firestore destructor is running on the AsyncQueue.